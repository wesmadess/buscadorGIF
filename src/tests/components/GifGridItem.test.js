import { shallow } from "enzyme";
import { GifGridItem } from "../../components/GifGridItem";

describe('Pruebas en <GifGridItem/>',()=>{

    const title = 'blase one punch man GIF';
    const url = 'https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=e69f9192jl7mju2h3onkv2ubbhnfyhosy9eevt5wj8cf4y8b&rid=giphy.gif';        
    const wrapper = shallow(<GifGridItem title={title} url={url} />)
    
    test('Debe mostrar el componente correctamente',()=>{        
        expect(wrapper).toMatchSnapshot();        
    })   
    
    test('Debe de tener un pÃ¡rrafo con el title',()=>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(title);
    })  

    test('Debe de tener la imagen igual al url y alt de los props',()=>{
        const img = wrapper.find('img');
        // console.log(img.prop('src'))
        expect(img.prop('src')).toBe(url);
        expect(img.prop('alt')).toBe(title);
    })  

    test('Debe de tener animate__fadeIn',()=>{
        const div = wrapper.find('div');
        const className = div.prop('className');
        expect(className.includes('animate__fadeIn')).toBe(true);
    })  

})