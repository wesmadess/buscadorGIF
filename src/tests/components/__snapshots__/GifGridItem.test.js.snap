// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem/> Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="blase one punch man GIF"
    src="https://media1.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=e69f9192jl7mju2h3onkv2ubbhnfyhosy9eevt5wj8cf4y8b&rid=giphy.gif"
  />
  <p>
    blase one punch man GIF
  </p>
</div>
`;
